changes/watch_darwin.go

```diff
@@ -0,0 +1,84 @@
+// heavily inspired by https://github.com/fsnotify/go.fsevents
```

Comment:
Thanks David. If you have ideas on how the FSEvents wrapper could be improved, please do chime in. Pull requests are also welcome.